gcc/rust/ast/rust-ast.h

```diff
@@ -1840,22 +1868,27 @@ class ASTFragment
   bool should_expand () const { return !is_error () && !nodes.empty (); }
 
   /**
-   * We need to make a special case for Expression fragments as only one
-   * Node will be extracted from the `nodes` vector
+   * We need to make a special case for Expression and Type fragments as only
+   * one Node will be extracted from the `nodes` vector
    */
 
-  bool is_expression_fragment () const
+  bool is_single_fragment (SingleASTNode::NodeType kind) const
```

Comment:
Not sure i like this interface.

I think it would be better to have a few helpers here:

```c++
bool is_single_fragment() const { return nodes.size() == 1; }

bool is_single_fragment_kind(SingleASTNode::NodeType kind) const { return is_single_fragment() && nodes.at(0).get_kind() == kind; }
```

Reply:
I think this function should actually be private, but I agree with you. I'll change the interface